test/e2e/storage/persistent_volumes-disruptive.go

```diff
@@ -230,10 +251,19 @@ func kubeletCommand(kOp kubeletOpt, c clientset.Interface, pod *v1.Pod) {
 	nodeIP, err := framework.GetHostExternalAddress(c, pod)
 	Expect(err).NotTo(HaveOccurred())
 	nodeIP = nodeIP + ":22"
-	sshResult, err := framework.SSH("sudo /etc/init.d/kubelet "+string(kOp), nodeIP, framework.TestContext.Provider)
-	Expect(err).NotTo(HaveOccurred())
+	sshResult, err := framework.SSH(fmt.Sprintf("sudo systemctl %s kubelet", string(kOp)), nodeIP, framework.TestContext.Provider)
+	Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error occured attempting to SSH to node %q", pod.Spec.NodeName))
 	framework.LogSSHResult(sshResult)
-
+	if sshResult.Code != 0 {
+		framework.Logf("Restarting kubelet service via `systemctl` failed.  Attempting restart via `service`.")
+		sshResult, err = framework.SSH(fmt.Sprintf("sudo service kubelet %s", string(kOp)), nodeIP, framework.TestContext.Provider)
```

Comment:
Why is `kOp` used here instead of "restart"? Eg. `kOp` could be "stop".
Also, `kOp` could have been passed in as "start" . Are you trying to ensure that the kubelet is always started in the event of an error above? If the error was for "stop" then in theory the kubelet was not stopped and does not need to be restarted. If the error was for "start" then, in theory, it won't start successfully the second time either -- unless some delay is needed, in which case an explicit sleep or poll loop should be used...